### ISSUE TITLE
Sprockets::FileNotFound raised for compass partial SCSS file in 4.0.0.beta4 

### ISSUE THREAD

**Issue by PikachuEXE** [rails/sprockets#420]
### Expected behavior

It should be able to resolve the file

### Actual behavior

Error raised since sprockets unable to find the file
`Sprockets::FileNotFound - couldn't find file '/users/pikachuexe/.rvm/gems/ruby-2.3.1/gems/compass-core-1.0.3/stylesheets/compass/_support.scss'`

### System configuration

- Sprockets version
  `4.0.0.beta4 `
- Ruby version
  `ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin14]`

Other related stuff:
`compass-core-1.0.3`

### Example App

To be created

**Comment by PikachuEXE** [comment]
Sample app created:
https://github.com/PikachuEXE/rails_sprockets_issue_420

**Comment by michaelherold** [comment]
**tl;dr**

The `chosen-rails` gem works on `sprockets-3.7.0` but not on `sprockets-4.0.0.beta4` or the `master` branch. I can't diagnose why the paths aren't loaded properly in Sprockets 4 but did verify that the sample works in 3.7.0 with no modifications. Below is a recounting of what I did for those who want to follow up. My hypothesis is that something changed with dependency path management in Sprockets 4 or the asset gem is not set up properly.

## Investigation

I tried looking into this and this is what I came up with:

* The sample app is attempting to `= require chosen-compass` in an SCSS file, which [I didn't think was supported](https://github.com/rails/sass-rails#important-note). I attempted switching it to `@import chosen-compass;` but that did not fix the issue.
* The sample app works with sprockets 3.7.0 and sprockets-rails 3.2.0 (changed the sprockets line to `gem "sprockets", "~> 3"` to test).
* This sample app _does not_ work with sprockets master.
* `compass-rails` has [a recent PR](https://github.com/Compass/compass-rails/pull/270) about Sprockets 4 that is not in a released version yet. Switching to `gem "compass-rails", github: "Compass/compass-rails"` does not fix the issue.
* `chosen-rails` appears to be [phasing out compass as a dependency](https://github.com/tsechingho/chosen-rails/commit/3fdcc15e8a598283b7ce9c6404e99222c5353a1a). This leads me to believe they've had issues with it.

I had a hunch that `chosen-rails` was the problem child, since it says that `compass-rails` is no longer a dependency, but clearly tries to require specific `compass` files. As such, I tried yanking those specific lines into the `page.scss` in the sample project and the page rendered properly. The `page.scss` looked like this:

```scss
@import "compass/css3/box-sizing";
@import "compass/css3/images";
@import "compass/css3/user-interface";
```

I then tried to add the `@import "chosen-compass";` line ... but that caused the original error to pop. I then tried to just do an `@import "chosen";` and that works. That means that the problem resides either in Sprockets or in chosen-rails' `chosen-compass` file.

Lastly, I tried re-adding `compass-rails` (and `compass` for thoroughness) as a dependency to `chosen-rails` and that did not fix the issue either.

This is where I got stuck.

## Hypothesis

Either `chosen-rails` is not setting paths to talk to `compass` or Sprockets is not properly setting up paths for libraries to talk to each other. Is this a bug or a change in Sprockets?

## Major version change or bug?

Obviously, since the example works in Sprockets 3.7.0, _something_ changed that doesn't let this gem work in version 4. Since it's a major version bump, changes like this are allowed, but it could just as well be a bug.

**Comment by senid231** [comment]
same issue

app/assets/config/manifest.js
```js
//= link_tree ../images
//= link application.js
//= link application.scss
```

app/assets/stylesheets/application.scss
``` scss
@import "bootstrap-sprockets";
@import "bootstrap";
@import "font-awesome";
```
page render fails
```
ActionView::Template::Error (Asset `application.css` was not declared to be precompiled in production.
Declare links to your assets in `app/assets/config/manifest.js`.

  //= link application.css
and restart your server):
    2: <html>
    3: <head>
    4:   <title><%= t('main_title') %></title>
    5:   <%= stylesheet_link_tag 'application', media: 'all' %>
    6:   <%= csrf_meta_tags %>
    7: </head>
    8: <body>
```

`ruby 2.3.1`
`rails 4.2.7`
`sprockets 4.0.0.beta4`

**Comment by PikachuEXE** [comment]
@senid231 
I think it's just due to you link `application.scss` instead of `application.css` inside `manifest.js`
Should be a different issue
This issue is about compass (I think)

**Comment by fauxparse** [comment]
I'm getting the same thing with `bourbon`. For now, a workaround similar to the one in #329 seems to work for me, at least in development, but it seems to be a problem with Sprockets, not just `chosen-rails` or `compass`.

**Comment by schneems** [comment]
Is this still an issue?

**Comment by PikachuEXE** [comment]
Yes the sample app I created still produces the same issue
